pkg/make/builder.go

```diff
@@ -0,0 +1,429 @@
+/* Package make provides some simple functions to handle build and go
+dependancies.
+
+We used to do this with gnumake rules, but as we added windows
+compatibility, we found make too limiting. Moving this into go allows
+us to write cleaner cross-platform code.
+*/
+
+package make
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"os/exec"
+	"os/user"
+	"path"
+	"path/filepath"
+	"runtime"
+	"strings"
+	"time"
+
+	"github.com/Masterminds/semver"
+	"github.com/go-kit/kit/log/level"
+	"github.com/kolide/kit/fs"
+	"github.com/kolide/launcher/pkg/contexts/ctxlog"
+	"github.com/pkg/errors"
+	"github.com/theupdateframework/notary/client"
+	"github.com/theupdateframework/notary/trustpinning"
+	"github.com/theupdateframework/notary/tuf/data"
+	"go.opencensus.io/trace"
+	"golang.org/x/sync/errgroup"
+)
+
+type Builder struct {
+	os           string
+	arch         string
+	static       bool
+	race         bool
+	stampVersion bool
+
+	goVer  *semver.Version
+	cmdEnv []string
+}
+
+type Option func(*Builder)
+
+func WithOs(o string) Option {
+	return func(b *Builder) {
+		b.os = o
+	}
+}
+
+func WithArch(a string) Option {
+	return func(b *Builder) {
+		b.arch = a
+	}
+}
+
+func WithStatic() Option {
+	return func(b *Builder) {
+		b.static = true
+	}
+}
+
+func WithRace() Option {
+	return func(b *Builder) {
+		b.race = true
+	}
+}
+
+func WithStampVersion() Option {
+	return func(b *Builder) {
+		b.stampVersion = true
+	}
+}
+
+// execCommand is a var to allow mocking in tesst
+var execCommand = exec.Command
```

Comment:
instead of these package globals, add `exec` as a field to Builder and swap it with a private option.

Reply:
Also, just eliminate the non context version. it's never needed.